browser/src/Services/Terminal.ts

```diff
@@ -15,26 +13,21 @@ export const activate = (
     configuration: Configuration,
     editorManager: EditorManager,
 ) => {
-    const getTerminalCommand = () => {
-        const terminalCommand = configuration.getValue("terminal.shellCommand", "")
-        return `term://${terminalCommand}`
-    }
-
-    const openTerminal = (openMode: Oni.FileOpenMode) => {
-        editorManager.activeEditor.openFile(getTerminalCommand(), { openMode })
+    const openTerminal = (vertical: boolean) => {
+        editorManager.activeEditor.neovim.command(`:${vertical ? "v" : ""}sp! | terminal`)
     }
 
     commandManager.registerCommand({
         command: "terminal.openInVerticalSplit",
         name: "Terminal: Open Vertical",
         detail: "Open a terminal emulator in a vertical split",
-        execute: () => openTerminal(Oni.FileOpenMode.VerticalSplit),
```

Comment:
Is there a reason we need to change this? (Or a case it's not working in today?)

The reason I ask is I'm working on a 'multiplexed' mode - you can enable it today with `"editor.split.mode": "oni"` (but it's still experimental). This lets you open editor windows in new tabs:
![multiplexed-windows](https://user-images.githubusercontent.com/13532591/38842309-192475ec-419e-11e8-9e1e-4b82443a6f94.gif)




So using the `openFile` API with that open mode will do `vsp`/`sp` if we aren't in that multiplexed mode... but if we are in that multiplexed mode, it'll use this new type of split.

(I actually showcase this in our intro video on the website, too! https://www.youtube.com/watch?v=uLpbxfq0IXM&feature=youtu.be&t=40s)

Reply:
The current implementation replaces the whole `shell` variable on vim side. We can just open `term://` (which is going to happen if `terminal.shellCommand` is null or eval `set shell` on the `terminal.shellCommand` configuration setting. This should be just a single line change for the whole PR.

Reply:
It's because I don't open a file here, i'm using a command, and oni's api currently doesn't have a wait to spawn a command with a FileOpenMode